### ISSUE TITLE
Type check for derived method of generic type

### ISSUE THREAD

**Issue by migel** [JukkaL/mypy#1197]
mypy doesn't complain about the following code. I expected to get an error on `B.f` because it returns `str` instead of `float`:

```python

import abc
from typing import Generic, TypeVar

TA = TypeVar('TA')
TB = TypeVar('TB')


class A(Generic[TA, TB],
        metaclass=abc.ABCMeta):

    @abc.abstractmethod
    def f(self, a: TA) -> TB:
        pass


class B(A[TA, float], Generic[TA]):

    def f(self, a: TA):
        return 'xyz'
```

**Comment by gvanrossum** [comment]
I guess this is by design. B.f's return type is Any (because it's not explicitly given) and a subclass is allowed to vary the return type of a method it overloads from the base class type by making it Any in the subclass. IOW this behaves the same way as if you had written `-> Any` in B.f, and the warning about a return type incompatible with that of a supertype is suppressed when the return type is Any. Try adding `-> str` to B.f and notice that you do get the error; replace that with `-> Any` and notice that you don't.

**Comment by migel** [comment]
Thanks, got it. If you don't specify a return type it assumes `Any`. It is consistent with the non generic case. The following also doesn't give an error:

```python
class C:
    def f(self) -> float:
        return 5.0


class D(C):
    def f(self):
        return 'x'
```
